astropy/stats/lombscargle/_lombscargle.py

```diff
@@ -0,0 +1,256 @@
+        """
+        t = np.asanyarray(self.t)
+        baseline = t.max() - t.min()
+        n_samples = t.size
+
+        df = 1. / baseline / samples_per_peak
+
+        if minimum_frequency is not None:
+            f0 = minimum_frequency
+        else:
+            f0 = 0.5 * df
+
+        if maximum_frequency is not None:
+            Nf = int(np.ceil((maximum_frequency - f0) / df))
+        else:
+            Nf = int(0.5 * samples_per_peak * nyquist_factor * n_samples)
+
+        return f0 + df * np.arange(Nf)
+
+    def autopower(self, method='auto', method_kwds=None,
+                  normalization='normalized', **kwargs):
+        """Compute Lomb-Scargle power at automatically-determined frequencies
+
+        Parameters
+        ----------
+        method : string (optional)
+            specify the lomb scargle implementation to use. Options are:
+
+            - 'auto': choose the best method based on the input
+            - 'fast': use the O[N log N] fast method. Note that this requires
+              evenly-spaced frequencies: by default this will be checked unless
+              ``assume_regular_frequency`` is set to True.
+            - 'slow': use the O[N^2] pure-python implementation
+            - 'cython': use the O[N^2] cython implementation. This is slightly
+              faster than method='slow', but much more memory efficient.
+            - 'chi2': use the O[N^2] chi2/linear-fitting implementation
+            - 'fastchi2': use the O[N log N] chi2 implementation. Note that this
+              requires evenly-spaced frequencies: by default this will be checked
+              unless ``assume_regular_frequency`` is set to True.
+            - 'scipy': use ``scipy.signal.lombscargle``, which is an O[N^2]
+              implementation written in C. Note that this does not support
+              heteroskedastic errors.
+
+        method_kwds : dict (optional)
+            additional keywords to pass to the lomb-scargle method
+        normalization : string (optional, default='normalized')
+            Normalization to use for the periodogram.
+            Options are 'normalized' or 'unnormalized'.
+        **kwargs :
+            additional keyword arguments will be passed to autofrequency()
+
+        Returns
+        -------
+        frequency, power : ndarrays
+            The frequency and Lomb-Scargle power
+        """
+        frequency = self.autofrequency(**kwargs)
+
+        power = lombscargle(self.t, self.y, self.dy,
+                            frequency=frequency,
+                            center_data=self.center_data,
+                            fit_bias=self.fit_bias,
+                            normalization=normalization,
+                            method=method, method_kwds=method_kwds,
+                            assume_regular_frequency=True)
+        return frequency, power
+
+    def power(self, frequency, normalization='normalized', method='auto',
```

Comment:
Initially I had it that way. The problem is that ``autopower()`` must return a frequency, while for ``power()`` it is redundant to return the frequency, because the user must provide it.

I don't like interfaces where the number of return values depends on how you call the function, so I chose to split this into two methods.

Would be open to hearing other opinions on this.